lab5 of python

cojlogkuu opened the pull request:
add lab5 and task description for it

cojlogkuu committed:
add lab 5

lab5_Bank.py contains class Bank

lab5_BankAccount.py contains class BankAccount

cojlogkuu committed:
lab5 description

write task description in README.md

Kapitoshka-a commented on README.md:
```diff
@@ -1 +1 @@
-# Labs-python
\ No newline at end of file
+Створити клас BankAccount, який дозволяє виконувати основні банківські операції: створити акаунт (повинен містити id, owner_name, balance), поповнити баланс і зняти готівку. Реалізуйте метод showAccount і додайте можливість поповнювати баланс на ціле і дробове число. Додайте клас Bank, що містить набір рахунків і дозволяє додавати/видаляти їх по id. Додати можливість сортування аккаунтів за балан ...
```
Add number of task

cojlogkuu replied:
added

Kapitoshka-a commented on lab5_BankAccount.py:
```diff
@@ -0,0 +1,89 @@
+class BankAccount:
+
+    __current_id = 0
```
capitalize the constant

cojlogkuu replied:
it is not a constant.  "__current_id" is changed when a new instance of the class is created

VitaliiPaliukh commented on lab5_Bank.py:
```diff
@@ -0,0 +1,24 @@
+    def __init__(self):
+        self.__accounts_list = dict()
+
+    def add_account(self, owner_name, balanceS = 0.0):
```
use balance instead

cojlogkuu replied:
fixed

Kapitoshka-a commented on lab5_BankAccount.py:
```diff
@@ -0,0 +1,89 @@
+        self.__balance -= self.check_money(money)
+        print(f'you have successfully withdrawn {money} from your account')
+
+    # @property
```
Delete coments

cojlogkuu replied:
deleted

VitaliiPaliukh commented on lab5_Bank.py:
```diff
@@ -0,0 +1,24 @@
+            raise KeyError('id was not found')
+
+    def sort_accounts(self):
+        return sorted(self.__accounts_list.values(), key=lambda x: x.balance, reverse = True)
```
потрібно використати "get_balance" тому що balance є приватним атрибутом в класі bankaccount

cojlogkuu replied:
вище прописаний метод гетера для __balance з декоратором property

Kapitoshka-a commented on lab5_BankAccount.py:
```diff
@@ -0,0 +1,89 @@
+    def balance(self, new_balance):
+        self.__balance = self.check_money(new_balance)
+
+    @staticmethod
```
remove static method and create such function in main file

cojlogkuu replied:
These methods are used to check that the input data is of the correct type. They should be in the class "BankAccount" as staticmethods because they will not be used anywhere else

VitaliiPaliukh commented on lab5_BankAccount.py:
```diff
@@ -0,0 +1,89 @@
+            raise TypeError('owner_name should be of type str')
+        elif not owner_name.replace(" ", "").isalpha():
+            raise ValueError('owner_name should consist of only letters')
+        elif len(owner_name.split()) != 2:
```
Треба позволити писати більше двох слів,щоб можна було написати по батькові

cojlogkuu replied:
тепер можна писати ще побатькові

Kapitoshka-a reviewed (commented):
add main file and implement the functionality there

VitaliiPaliukh reviewed (commented):
поправити

cojlogkuu committed:
Add number of task

cojlogkuu committed:
Fixed code in "lab5_Bank" and "lab5_BankAccount"

Corrected code according to the advice that was in the reviews of Kapitoshka-a and VitaliiPaliukh

cojlogkuu committed:
added file main.py

according to Kapitoshka's review, added  file main.py that contains the code for lab5
